brokers/artifacts/broker.go

```diff
@@ -58,122 +57,125 @@ func (b *Broker) Start(pluginFQNs []model.PluginFQN, defaultRegistry string) err
 	b.PubStarted()
 	b.PrintInfo("Starting plugin artifacts broker")
 
-	b.cleanupPluginsDirectory()
 	pluginMetas, err := utils.GetPluginMetas(pluginFQNs, defaultRegistry, b.ioUtils)
 	if err != nil {
 		return b.fail(fmt.Errorf("Failed to download plugin meta: %s", err))
 	}
-	b.PrintInfo("Downloading plugin extensions")
 
 	err = utils.ResolveRelativeExtensionPaths(pluginMetas, defaultRegistry)
 	if err != nil {
 		return b.fail(err)
 	}
 
-	for _, meta := range pluginMetas {
-		err = b.ProcessPlugin(meta)
+	requestedPlugins := convertMetasToPlugins(pluginMetas)
+
+	toInstall := b.SyncPlugins(requestedPlugins)
+
+	for _, plugin := range toInstall {
+		err = b.ProcessPlugin(&plugin)
 		if err != nil {
 			return b.fail(err)
 		}
 	}
 
+	err = b.writeInstalledPlugins(toInstall)
+	if err != nil {
+		b.PrintInfo("WARN: Failed to log installed plugins: %s", err)
+	}
+
 	b.PrintInfo("All plugin artifacts have been successfully downloaded")
 	b.PubDone("")
-
 	return nil
 }
 
-func (b *Broker) cleanupPluginsDirectory() {
-	b.PrintInfo("Cleaning /plugins dir")
-	files, err := b.ioUtils.GetFilesByGlob(filepath.Join("/plugins", "*"))
+// ProcessPlugin downloads all undownloaded plugin extensions and places the
+// relevant artifacts in their appropriate location in the /plugins directory.
+// If a plugin already has artifacts downloaded for a given extension, that
+// extension is skipped.
+func (b *Broker) ProcessPlugin(plugin *model.CachedPlugin) error {
+	workDir, err := b.ioUtils.TempDir("", "vscode-extension-broker")
```

Comment:
Just wondering if it would be better to rename the folder to `artifacts-broker` since there is no explicit VSCode related logic in code, everything in name in a general manner - Plugin, Extension.

Reply:
:+1: this is a holdover from before the refactor, will update.

Reply:
:+1:

Reply:
> :+1: 

:+1: 

;)